identity-service/src/routes/trackListens.js

```diff
@@ -436,4 +437,48 @@ module.exports = function (app) {
       listenMap
     })
   }))
+
+  /**
+   * Gets user listen records in bulk.
+   * 
+   * GET query parameters:
+   *  startTime (string) the start time to fetch listens from (in unix seconds format)
+   *  limit (number) the number of records to fetch from each user listens and anonymous listens
+   */
+  app.get('/listens/bulk', handleResponse(async (req, res) => {
+    let { startTime, limit } = req.query
+    if (!startTime || !limit) {
+      return errorResponseBadRequest('Please provide a startTime and limit')
+    }
+
+    limit = Math.min(limit, 1000)
```

Comment:
Couple things, limit is a string here, but Math.mill with convert to number but maybe run parseInt on it so its not a decimal/negative or whatever could also check that limit is > 0. Not actually too worried about it. could leave as is.

Reply:
👍